src/vs/workbench/api/common/extHostEditorTabs.ts

```diff
@@ -8,43 +8,65 @@ import { IEditorTabDto, IExtHostEditorTabsShape } from 'vs/workbench/api/common/
 import { URI } from 'vs/base/common/uri';
 import { Emitter, Event } from 'vs/base/common/event';
 import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { ViewColumn } from 'vs/workbench/api/common/extHostTypes';
 
 export interface IEditorTab {
-	name: string;
-	group: number;
-	resource: vscode.Uri
+	label: string;
+	viewColumn: ViewColumn;
+	resource?: vscode.Uri
 	isActive: boolean
 }
 
 export interface IExtHostEditorTabs extends IExtHostEditorTabsShape {
 	readonly _serviceBrand: undefined;
 	tabs: readonly IEditorTab[];
-	onDidChangeTabs: Event<void>;
+	activeTab: IEditorTab | undefined;
+	onDidChangeActiveTab: Event<IEditorTab | undefined>;
+	onDidChangeTabs: Event<IEditorTab[]>;
 }
 
 export const IExtHostEditorTabs = createDecorator<IExtHostEditorTabs>('IExtHostEditorTabs');
 
 export class ExtHostEditorTabs implements IExtHostEditorTabs {
 	readonly _serviceBrand: undefined;
 
-	private readonly _onDidChangeTabs = new Emitter<void>();
-	readonly onDidChangeTabs: Event<void> = this._onDidChangeTabs.event;
+	private readonly _onDidChangeTabs = new Emitter<IEditorTab[]>();
+	readonly onDidChangeTabs: Event<IEditorTab[]> = this._onDidChangeTabs.event;
+
+	private readonly _onDidChangeActiveTab = new Emitter<IEditorTab | undefined>();
+	readonly onDidChangeActiveTab: Event<IEditorTab | undefined> = this._onDidChangeActiveTab.event;
 
 	private _tabs: IEditorTab[] = [];
+	private _activeTab: IEditorTab | undefined;
 
 	get tabs(): readonly IEditorTab[] {
 		return this._tabs;
 	}
 
+	get activeTab(): IEditorTab | undefined {
+		return this._activeTab;
+	}
+
 	$acceptEditorTabs(tabs: IEditorTabDto[]): void {
+		this._activeTab = undefined;
 		this._tabs = tabs.map(dto => {
+			// If we iterate through and find the active tab also set that
+			if (dto.isActive) {
+				this._activeTab = {
+					label: dto.label,
+					viewColumn: dto.viewColumn,
+					resource: URI.revive(dto.resource),
+					isActive: true
+				};
+			}
 			return {
-				name: dto.name,
-				group: dto.group,
+				label: dto.label,
```

Comment:
Make sure to enforce the `readonly`-ness of things, use `Object.freeze` to ensure no extension can modify API objects